src/modules/material-requests/services/material-requests.service.ts

```diff
@@ -99,6 +102,24 @@ export class MaterialRequestsService {
 			};
 		}
 
+		// Check if maintainer id is not empty
+		// If so check where object already has an entry for the maintainer schema identifier
+		// If so push the user maintainerId to the array
+		// If not add an entry for maintainer schema identifier
+		if (!isEmpty(parameters.userMaintainerId)) {
+			if (where?.object?.maintainer?.schema_identifier) {
+				where.object.maintainer.schema_identifier._in.push(parameters.userMaintainerId);
+			} else {
+				where.object = {
+					maintainer: {
+						schema_identifier: {
+							_in: [parameters.userMaintainerId],
+						},
+					},
+				};
+			}
+		}
```

Comment:
PP: check access stuff in the controller and just pass a maintainerIds list to the service.

eg:
in controller check if user is cp_admin
if yes => set maintainerIds to user's maintainer id
if no => keep maintainer ids of the request params

Service just receives a simple maintainerIds list